models/User.js

```diff
@@ -0,0 +1,31 @@
+const mongoose = require('mongoose');
+const Habit = require('./Habit');
+
+const { Schema } = mongoose;
+
+// Create User schema
+const userSchema = new Schema({
+	username: {
+		type: String,
+		required: true,
+	},
+	email: {
+		type: String,
+		required: true,
+	},
+	password: {
+		type: String,
+		required: true,
+	},
+	// to record the date when the user registered
+	date: {
+		type: Date,
+		default: Date.now,
+	},
+	// habits assigned to this user
+	habits: {
+		type: [Habit],
```

Comment:
I'm not sure that this will work. I think that you need `ref` to Habit model here. Maybe something like this:

```js
type: [mongoose.Schema.Types.ObjectId],
required: true,
ref: 'Habit',
```

Reply:
I was thinking something like this,
https://mongoosejs.com/docs/subdocs.html

but this is perhaps better, to keep them separate